src/ios/CameraViewController.m

```diff
@@ -37,9 +36,89 @@ @interface CameraViewController ()<AVCaptureVideoDataOutputSampleBufferDelegate>
 
 @end
 
+// muchas gracias to: https://stackoverflow.com/a/6672628/1856221
+@implementation UIImage (NegativeImage)
+
+- (UIImage *)negativeImage
+{
+    // get width and height as integers, since we'll be using them as
+    // array subscripts, etc, and this'll save a whole lot of casting
+    CGSize size = self.size;
+    int width = size.width;
+    int height = size.height;
+    
+    // Create a suitable RGB+alpha bitmap context in BGRA colour space
+    CGColorSpaceRef colourSpace = CGColorSpaceCreateDeviceRGB();
+    unsigned char *memoryPool = (unsigned char *)calloc(width*height*4, 1);
```

Comment:
UHG
what is this monstrosity of a language?

Reply:
Well, I don't know a single person that likes objective c. And I think I don't know a language which is so widespread (or even mandatory) and this ugly. 🙈🙉🙊